internal/data/data_manager.go

```diff
@@ -539,3 +540,42 @@ func (dm *dataManager) WriteNewMessage(ctx context.Context, newMsg *NewMessage)
 func (dm *dataManager) WaitStop() {
 	dm.messageWriter.close()
 }
+
+func (dm *dataManager) DeleteData(ctx context.Context, dataID string) error {
+	id, err := fftypes.ParseUUID(ctx, dataID)
+	if err != nil {
+		return err
+	}
+
+	data, err := dm.database.GetDataByID(ctx, dm.namespace.Name, id, false)
+	if err != nil {
+		return err
+	}
+	if data == nil {
+		return i18n.NewError(ctx, coremsgs.Msg404NoResult)
+	}
+	if data.Blob != nil && data.Blob.Hash != nil {
+
+		blob, err := dm.database.GetBlob(ctx, dm.namespace.Name, data.ID, data.Blob.Hash)
```

Comment:
I believe we determined there could be multiple blobs at this point @nguyer 

e.g. I was expecting to see:

```go
blobs, err := dm.database.GetBlobs(ctx, db.namespace.Name, fb.And(fb.Eq("data_id",data.ID), fb.Eq(data.Blob.Hash))
```

I'm not sure it even makes sense to have a `GetBlob` database-level action, because the only unique key for a record is the `Sequence` key (which we correctly use in delete).

Reply:
The specific case I'm thinking of, is when you receive the same data, with the same ID/Hash from multiple location in your business next, in multiple different messages.

Reply:
Thanks for working that all through 👍